sci-geosciences/GeographicLib/GeographicLib-1.51-r1.ebuild

```diff
@@ -0,0 +1,110 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+PYTHON_COMPAT=( python3_{7,8,9} )
+
+inherit cmake distutils-r1
+
+MY_PN="${PN,,}"
+
+DESCRIPTION="C++ library for converting geographic coordinate systems"
+HOMEPAGE="https://sourceforge.net/projects/geographiclib/"
+SRC_URI="mirror://sourceforge/${MY_PN}/${P}.tar.gz"
+LICENSE="MIT"
+
+# To-Do:find out why this doesn't work
+# 105/167 Test #105: GeodSolve82 ......................***Failed
+# Required regular expression not found. Regex=[20\.0010* 0\.0* 180\.0* 20\.0010* 0\.0* 180\.0* 0\.0* 0\.0* 0\.0* 1\.0* 1\.0* -?0
+RESTRICT="test"
+
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="doc examples python"
+
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
+RDEPEND=""
+DEPEND="
+	${RDEPEND}
+	doc? (
+		>=app-doc/doxygen-1.8.7
+		>=dev-lang/perl-5.26.1-r1
+		>=dev-python/sphinx-1.6.3-r2
+		>=sys-apps/util-linux-2.31
+	)
+"
+
+distutils_enable_tests setup.py
+# there are additional docs in the python dir
+distutils_enable_sphinx python/doc
+
+src_prepare() {
+	cmake_src_prepare
+
+	if use python; then
+		pushd "python" || die
+		distutils-r1_python_prepare_all
+		popd || die
+	fi
+}
+
+src_configure() {
+	export GEODATAPATH="/usr/share/${MY_PN}"
+
+	local mycmakeargs=(
+		-DGEOGRAPHICLIB_DOCUMENTATION=$(usex doc ON OFF)
+		-DGEOGRAPHICLIB_LIB_TYPE="SHARED"
+		-DGEOGRAPHICLIB_DATA="${GEODATAPATH}"
+	)
+	cmake_src_configure
+}
+
+src_compile() {
+	cmake_src_compile
+
+	if use python; then
+		pushd "python" || die
+		python_foreach_impl distutils-r1_python_compile
+		popd || die
+		use doc && build_sphinx python/doc
+	fi
+}
+
+src_test() {
+	# Only 1 failing test in the C code, python passes for me
+	cmake_src_test
+
+	if use python; then
+		pushd "python" || die
+		python_foreach_impl python_test
+		popd || die
+	fi
+}
+
+src_install() {
+	cmake_src_install
+
+	# remove python things added by the cmake_src_install function
+	# these are installed in the wrong python dir
+	rm -rf "${D}/usr/$(get_libdir)/python" || die
```

Comment:
You might be able to fix this path (and the doc one) in the configure phase with an appropriate `-D....=...` option.

Reply:
which one? `DOCDIR` doesn't get picked up by cmake

Reply:
even if I fix the python install path, python modules needs to be installed multiple times in multiple paths

Reply:
> even if I fix the python install path, python modules needs to be installed multiple times in multiple paths

Good point, never mind then